### PR TITLE
chore: upgrade postgrest tests image add some more test data

### PR DESCRIPTION
**Pull request by avallete**
## What kind of change does this PR introduce?

- Upgrade the postgrest docker image version to match recent one used in supabase
- Add some more data to the tests to assert postgrest runtime behavior on some corner case
- Update all data snapshots in the tests

## What is the current behavior?

Please link any relevant issues here.

## What is the new behavior?

Feel free to include screenshots if it includes visual changes.

## Additional context

Add any other context or screenshots.

### CODE REVIEW

File: test/db/docker-compose.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 version: '3'
 services:
   rest:
-    image: postgrest/postgrest:v11.2.2
+    image: postgrest/postgrest:v12.2.0
     ports:
```
Comment (avallete) [inline test/db/docker-compose.yml:7]: That's the important part, everything else is tests updates due to changes in 01-dummy-data.sql

**Review by soedirgo — APPROVED** [review]
